File: Python/cone/cone.py
# Starten in der Anaconda-Shell mit
#    python cone.py
import vtk


coneSource = vtk.vtkConeSource()
# coneSource.SetResolution(60)
# coneSource.SetCenter(-2,0,0)

# Mapper und Actor erzeugen und verbinden
mapper = vtk.vtkPolyDataMapper()
mapper.SetInputConnection(coneSource.GetOutputPort())
actor = vtk.vtkActor()
actor.SetMapper(mapper)

# Hintergrundfarbe des Fensters mischen
colors = vtk.vtkNamedColors()
colors.SetColor("BkgColor", [26, 51, 77, 255])

# Renderer, Fenster und Interactor
renderer = vtk.vtkRenderer()
renderer.AddActor(actor)
renderer.SetBackground(colors.GetColor3d("BkgColor"))  # Background color dark red

renderWindow = vtk.vtkRenderWindow()
renderWindow.SetPosition(100,100)
renderWindow.SetSize(500,500)
renderWindow.AddRenderer(renderer)

renderWindowInteractor = vtk.vtkRenderWindowInteractor()
renderWindowInteractor.SetRenderWindow(renderWindow)

renderWindow.Render()
renderWindowInteractor.Start()

File: CC/cone/Cone.cpp
/*
 * Erstes "Hello-World" für die Verwendung von VTK.
 * Wir erzeugen einen Kegel und stellen diesen
 * in einem OpenGL-Fenster dar.
 * Entspricht dem VTK-Beispiel "Cone".
 */
#include <vtkConeSource.h>
#include <vtkPolyData.h>
#include <vtkSmartPointer.h>
#include <vtkPolyDataMapper.h>
#include <vtkActor.h>
#include <vtkRenderWindow.h>
#include <vtkRenderer.h>
#include <vtkRenderWindowInteractor.h>

int main(int, char *[])
{
  // Kegel erzeugen
  vtkSmartPointer<vtkConeSource> coneSource =
    vtkSmartPointer<vtkConeSource>::New();

  // Mapper und Actor erzeugen und Kegel übergeben
  vtkSmartPointer<vtkPolyDataMapper> mapper =
    vtkSmartPointer<vtkPolyDataMapper>::New();
  mapper->SetInputConnection(coneSource->GetOutputPort());

  vtkSmartPointer<vtkActor> actor =
    vtkSmartPointer<vtkActor>::New();
  actor->SetMapper(mapper);

  // Renderer, Fenster und Interactor erzeugen
  vtkSmartPointer<vtkRenderer> renderer =
    vtkSmartPointer<vtkRenderer>::New();
  vtkSmartPointer<vtkRenderWindow> renderWindow =
    vtkSmartPointer<vtkRenderWindow>::New();
  renderWindow->AddRenderer(renderer);
  vtkSmartPointer<vtkRenderWindowInteractor> renderWindowInteractor =
    vtkSmartPointer<vtkRenderWindowInteractor>::New();
  renderWindowInteractor->SetRenderWindow(renderWindow);

  // Actor an Renderer übergeben, Hintergrundfarbe setzen
  renderer->AddActor(actor);
  renderer->SetBackground(0.3, 0.3, 0.3);

  // Anzeigen und interagieren
  renderWindow->Render();
  renderWindowInteractor->Start();
  
  return EXIT_SUCCESS;
}
